velox/dwio/dwrf/reader/ReaderBase.h

```diff
@@ -48,9 +48,12 @@ class FooterStatisticsImpl : public dwio::common::Statistics {
 
   virtual ~FooterStatisticsImpl() override = default;
 
-  virtual const dwio::common::ColumnStatistics& getColumnStatistics(
+  virtual const dwio::common::ColumnStatistics* getColumnStatistics(
       uint32_t columnId) const override {
-    return *colStats_.at(columnId);
+    if (!colStats_.empty()) {
```

Comment:
This also seems wrong.  Even if column statistics are not written, the number of columns here should still be correct and they are all populated with some empty statistics that will pass all the filter.

Reply:
> This also seems wrong. Even if column statistics are not written, the number of columns here should still be correct and they are all populated with some empty statistics that will pass all the filter.

If the column stats are not written, this colStats_ vector would be empty. In this case a nullptr should be returned.